test/CMakeLists.txt

```diff
@@ -18,6 +18,9 @@ endif()
 configure_file(ArborX_EnableDeviceTypes.hpp.in ${CMAKE_CURRENT_BINARY_DIR}/ArborX_EnableDeviceTypes.hpp @ONLY)
 
 find_package(Boost 1.67.0 REQUIRED COMPONENTS unit_test_framework)
+if(${Boost_VERSION_STRING} VERSION_GREATER_EQUAL 1.69.0)
```

Comment:
According to CMake [boost package](https://gitlab.kitware.com/cmake/cmake/blob/master/Modules/FindBoost.cmake#L35):
```
  Boost_VERSION_STRING   - Boost version number in x.y.z format
  Boost_VERSION          - if CMP0093 NEW => same as Boost_VERSION_STRING
                           if CMP0093 OLD or unset => same as Boost_VERSION_MACRO
```
if one uses `Boost_VERSION` it may mean different things depending on CMake version (pre-3.15, or later). I switched to `Boost_VERSION_STRING` so that it works properly.

Reply:
This restriction is only necessary if `CUDA` is enabled as backend, isn't it?